util/storage/prom-pushgateway/pushgateway.go

```diff
@@ -0,0 +1,82 @@
+package prom_pushgateway
+
+import (
+	"github.com/bblfsh/performance/util"
+	"github.com/bblfsh/performance/util/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/src-d/envconfig"
+	"golang.org/x/tools/benchmark/parse"
+)
+
+// TODO logs, comments, commands doc
+
+// Kind is a string that represents prometheus pushgateway
+const Kind = "prom"
+
+type metrics map[string]*prometheus.SummaryVec
+
+type promClient struct {
+	*push.Pusher
+}
+
+type promConfig struct {
+	Address string
+	Job     string
+}
+
+func init() {
+	storage.Register(Kind, NewClient)
+}
+
+// NewClient is a constructor that uses environment variables to create a wrapper around *push.Pusher,
+func NewClient() (storage.Client, error) {
+	var promConfig promConfig
+	if err := envconfig.Process("prom", &promConfig); err != nil {
+		return nil, err
+	}
+	return &promClient{push.New(promConfig.Address, promConfig.Job)}, nil
+}
+
+// Dump stores given benchmark results with tags to prometheus pushgateway
+func (c *promClient) Dump(tags map[string]string, benchmarks ...*parse.Benchmark) error {
+	labels, values := util.SplitStringMap(tags)
+	labels = append([]string{"name"}, labels...)
+
+	metrics := getMetrics(labels)
+	for _, b := range benchmarks {
+		tmpValues := append([]string{util.ParseBenchmarkName(b.Name)}, values...)
+
+		metrics[storage.PerOpSeconds].WithLabelValues(tmpValues...).Observe(float64(b.NsPerOp / 1e9))
```

Comment:
here I suggest to use `time.Duration(b.NsPerOp).Seconds()`

Reply:
Done